pkg/webhooks/admission/queues/validate/validate_hierarchical_queue.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2024 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validate
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	whv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/klog/v2"
+
+	schedulingv1beta1 "volcano.sh/apis/pkg/apis/scheduling/v1beta1"
+	"volcano.sh/volcano/pkg/webhooks/router"
+	"volcano.sh/volcano/pkg/webhooks/schema"
+	"volcano.sh/volcano/pkg/webhooks/util"
+)
+
+func init() {
+	router.RegisterAdmission(hierarchyService)
+}
+
+var hierarchyService = &router.AdmissionService{
+	Path:   "/hierarchicalqueues/validate",
+	Func:   HierarchicalQueues,
+	Config: config,
+	MutatingConfig: &whv1.MutatingWebhookConfiguration{
+		Webhooks: []whv1.MutatingWebhook{{
+			Name: "validatehierarchicalqueues.volcano.sh",
+			Rules: []whv1.RuleWithOperations{
+				{
+					Operations: []whv1.OperationType{whv1.Create, whv1.Update, whv1.Delete},
+					Rule: whv1.Rule{
+						APIGroups:   []string{schedulingv1beta1.SchemeGroupVersion.Group},
+						APIVersions: []string{schedulingv1beta1.SchemeGroupVersion.Version},
+						Resources:   []string{"queues"},
+					},
+				},
+			},
+		}},
+	},
+}
+
+// HierarchicalQueues validate queues when capacity plugin enables hierarchy.
+func HierarchicalQueues(ar admissionv1.AdmissionReview) *admissionv1.AdmissionResponse {
```

Comment:
What's the consideration here?
Here will also cause the problem that queue controller keep retrying.

Reply:
This won't cause the queue controller to keep retrying. The issue you mentioned only occurs when performing operations related to opening or closing queues. However, there might be some scenarios that I haven't considered. Could you provide a specific example to illustrate such a case?